docs/prerequisites.md

```diff
@@ -0,0 +1,76 @@
+# Deployment of Metal3 on vanilla K8s cluster
+To deploy metal3-components on vanilla K8s cluster, the following prerequisites
+have to be met:
+1. **Ironic should have access to layer-2 network for provisioning.**
+2. **Firewall is configured properly**
+3. **Webserver container containing  node images is running and reachable**
+4. **Ironic-bmo-configmap is populated correctly**
+
+We elaborate these points in detail here:
+1. Ironic should have access to layer-2 network for provisioning. It should be
+running on **host** networking. And on top of that the network should be
+configured so that nodes can reach the networking service for DHCP, PXE boot .
+It is also required to provide ironic instrospection with the MAC address(es) of
+ each node that ironic is provisioning so that it can pass this information to
+ networking service for DHCP and PXE boot configuration.
+
+2. Firewall should be configured to allow the required traffic to pass through.
+The following traffic should be allowed at least:
+  * ARP
+  * DHCP
+  * VRRP
+  * ICMP
+  * HTTP towards internal and external webserver
+  * Ports for the above mentioned services and for `Ironic-IPA`.   
+
+3. The webserver container containing node images volume should be running and
+reachable. It is called the `httpd-infra` container in metal3-dev-env, which
+runs on ironic image and contains the node images (OS images). It also caches
+a few other packages which are required for the second webserver `ironic-httpd`
+ which runs inside the cluster in `Baremetal Operator` deployment. The following
+ tree structure shows an example of the volume mounted in the external webserver
+ container with the required node images and other cached images:
+
+```ini
+/shared/
+├── html
+│   ├── dualboot.ipxe
+│   ├── images
+│   │   ├── bionic-server-cloudimg-amd64.img
+│   │   ├── bionic-server-cloudimg-amd64.img.md5sum
+│   │   ├── ironic-python-agent-1862f800-59e2c9cab7e95
+│   │   │   ├── ironic-python-agent.initramfs
+│   │   │   ├── ironic-python-agent.kernel
+│   │   │   ├── ironic-python-agent.tar
+│   │   │   └── ironic-python-agent.tar.headers
+│   │   ├── ironic-python-agent.initramfs -> ironic-python-agent-1862f800-59e2c9cab7e95/ironic-python-agent.initramfs
+│   │   ├── ironic-python-agent.kernel -> ironic-python-agent-1862f800-59e2c9cab7e95/ironic-python-agent.kernel
+│   │   └── ironic-python-agent.tar.headers -> ironic-python-agent-1862f800-59e2c9cab7e95/ironic-python-agent.tar.headers
+│   ├── inspector.ipxe
+│   └── uefi_esp.img
+└── tmp
+
+```
+
+4. The environments variables defined in `ironic-bmo-configmap` required for
```

Comment:
if deploying ironic locally (with the script in the tools folder) then the configmap is the CI one by default